### PR TITLE
Add documentation uploading script

### PR DESCRIPTION
**Pull request by ghoshbishakh**
upload_docs.py script builds documentation for the current version and uploads it to gh-pages branche of the dipy_web repository

### CODE REVIEW

**Comment by sahmed95** [comment]
@ghoshbishakh 
Hi, looks like you're working on documentation. Can you please have a look at #1022. 

Running "make ipynbexamples" from the /doc folder generates all the "valid_examples" as .ipynb files in the folder "ipython_notebooks". The notebooks aren't executed though and all that this script does is convert the .py files in the "examples" directory to Ipython notebooks. This is pertaining to #720

**Comment by ghoshbishakh** [comment]
@sahmed95 So we have to include it in the documentation somehow?

**Comment by sahmed95** [comment]
As @arokem mentioned in #720 

>> And vice versa: enable writing examples in IPython notebooks, which then get converted into the compliant python + rst to get built into the documentation.
 :+1: 1  

>> However, I would prefer that we automate this, such that the notebooks get automatically generated when we build the documentation. That is, rather than committing all these notebooks to the repository, we would have a script that converts the python examples to jupyter notebooks, and uploads them together with the python example scripts, so they can be downloaded. Even better, a script that would automatically create/update a gist on github and create the nbviewer page would be fantastic.

>> Would it be possible to do this as a separate instruction in the doc Makefile, rather than part of running the examples?

>> For now, all I am suggesting is to automate things so that when we run make html it not only produces the .rst versions of the examples, but also the .ipynb versions. 

So I wrote the script such that running ```make ipynbexamples``` generates the ipython notebooks.

**Comment by sahmed95** [comment]
I guess, the target is to build the ipython notebooks as one runs ```make html``` and produce the .rst versions of the examples and .ipynb notebooks and upload them. 

Currently ipython notebooks are converted by running ```make ipynbexamples```. So I guess you can help out by making this a part of ```make html``` and uploading the ipynb files as well after converting to .rst perhaps.

**Comment by ghoshbishakh** [comment]
Okey.. understood. I will start working on it.

**Comment by sahmed95** [comment]
Thanks for taking this up.

**Comment by ghoshbishakh** [comment]
@Garyfallidis @jchoude In Scikit image and Django website the docs are listed according to release versions and there is one version called "dev".
So in our case should I check if the version number has ".dev" at the end and put it as the latest dev version?

**Comment by jchoude** [comment]
Yes I think that would be the bvest. However, I would default to showing the doc for te latest release as the default, since that is what most users will install.

**Comment by jchoude** [comment]
Hi @ghoshbishakh this is still set as WIP. What's the status?

**Comment by ghoshbishakh** [comment]
@jchoude It is actually complete. May be we can merge it and make separate PRs later for bug fixes?

**Comment by ghoshbishakh** [comment]
@Garyfallidis can this be merged? If further changes are required in future then I will make separate PRs.

File: doc/upload_docs.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+def sh(cmd):
+    """Execute command in a subshell, return status code."""
+    print("--------------------------------------------------")
+    print("Executing: %s" % (cmd, ))
+    print("--------------------------------------------------")
+    return check_call(cmd, shell=True)
+
+# paths
+docs_repo_path = "_build/docs_repo"
+docs_repo_url = "git@github.com:ghoshbishakh/dipy_web.git"
```
Comment (jchoude) [inline doc/upload_docs.py]: Should probably point to nipy/dipy_web.git

**Comment by jchoude** [comment]
Except for my last comment, for me this is +1

**Comment by ghoshbishakh** [comment]
@jchoude I have changed the repository url

**Comment by jchoude** [comment]
Seems good to me. @ghoshbishakh Thanks.

@arokem @Garyfallidis anyone wants to have a quick look? Else I'll merge by the end of the week.

**Comment by jchoude** [comment]
@ghoshbishakh Two things

1) Can you rebase on master, please?

2) Upon final review and testing, I realize that changing the repo_url to point to the normal dipy_web repo causes an issue: the gh-pages branch doesn't currently exist there. Can you please fix?

**Comment by ghoshbishakh** [comment]
@jchoude I have rebased it on master. And added some error handeling.

**Comment by Garyfallidis** [comment]
This looks good. Thanks for addressing the comments!

**Pull request merged**